### DISCUSSION TITLE
Enhancement: Admin site

### DISCUSSION THREAD

**Discussion by MoBoo** [litestar-org/litestar#1721] [Ideas]
### Summary

Developing and including a built-in admin site for litestar, similar to Django's admin site.

Django's admin site is often the usp when deciding on which web framework to use.

There is already a similar project for SQLAlchemy: https://github.com/aminalaee/sqladmin, that could be adapted (?)

### Basic Example

https://docs.djangoproject.com/en/4.2/intro/tutorial02/#start-the-development-server
https://docs.djangoproject.com/en/4.2/ref/contrib/admin/

**Comment by Goldziher** [comment]
Hi @MoBoo. 

While I used and appreciate the Django Admin - litestar is not meant to be a batteries included framework on the scale of Django. We aim to be somewhere between Flask and Django while keeping things nimble and splitting as much as possible to contrib. 

This could be a great third party package, but I personally dont see us adding this to litestar. Perhaps the other maintainers feel differently though. Lets see.

**Comment by sinisaos** [comment]
@MoBoo If you are not using SqlAlchemy, you can try [Piccolo ORM](https://github.com/piccolo-orm/piccolo) which has an integrated [admin UI](https://github.com/piccolo-orm/piccolo_admin). Litestar has the option of [mounting the Piccolo admin UI](https://github.com/piccolo-orm/piccolo/blob/master/piccolo/apps/asgi/commands/templates/app/_litestar_app.py.jinja#LL18C1-L21C78) (a separate ASGI application) and the Piccolo integration via a [plugin](https://docs.litestar.dev/latest/usage/plugins/piccolo-orm.html) (which is optional as the plugin is basically a [create_pydantic_model](https://piccolo-orm.readthedocs.io/en/latest/piccolo/serialization/index.html) wrapper). Feel free to try Piccolo, it might suit your needs. I hope that is useful.

**Comment by provinzkraut** [comment]
A built-in, general purpose admin site similar to Django's is - nearly - impossible. Mainly because we don't have that tight coupling to an ORM Django has, which means we'd need to either restrict it to SQLAlchemy, find an abstraction over all the possible ORMs, and / or greatly reduce the feature set.

I do think this could possibly be a great first-party package, with a reduced scope maybe.
